VideoLocker/src/main/java/org/edx/mobile/model/api/CourseEntry.java

```diff
@@ -248,22 +252,80 @@ public boolean isGroupAvailable(SocialUtils.SocialType type) {
 
     }
 
-    public String getDescription(Context context){
+    public String getDescription(Context context, boolean withStartDate) {
         StringBuilder detailBuilder = new StringBuilder();
-        if ( getOrg() != null){
-            detailBuilder.append( getOrg());
+
+        if (!TextUtils.isEmpty(org)) {
+            detailBuilder.append(org);
         }
-        if ( getNumber() != null) {
-            if (detailBuilder.length() > 0){
+
+        if (!TextUtils.isEmpty(number)) {
+            if (detailBuilder.length() > 0) {
                 detailBuilder.append(" | ");
             }
-            detailBuilder.append( getNumber());
+            detailBuilder.append(number);
 
         }
-        //https://openedx.atlassian.net/browse/MA-858
-        //we remove the ending data for now
+
+        if (withStartDate) {
+            if (detailBuilder.length() > 0) {
+                detailBuilder.append(" | ");
+            }
+
+            detailBuilder.append(getFormattedStartDate(context));
+        }
+
+        if (detailBuilder.length() > 0) {
+            int secondlastCharIndex = detailBuilder.length() - 2;
+            if (detailBuilder.charAt(secondlastCharIndex) == '|') {
+                detailBuilder.delete(secondlastCharIndex, detailBuilder.length());
+            }
+        }
 
         return detailBuilder.toString();
     }
 
+    @SuppressLint("SimpleDateFormat")
+    public String getFormattedStartDate(Context context){
+        StringBuilder dateBuilder = new StringBuilder();
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("MMMM dd");
```

Comment:
@aleffert: I don't think `SimpleDateFormat` is meant for localization. The warning that is being suppressed here stated:

> Almost all callers should use `getDateInstance()`, `getDateTimeInstance()`, or
`getTimeInstance()` to get a ready-made instance of `SimpleDateFormat` suitable
for the user's locale. The main reason you'd create an instance this class
directly is because you need to format/parse a specific machine-readable
format, in which case you almost certainly want to explicitly ask for `US` to
ensure that you get ASCII digits (rather than, say, Arabic digits).

> Therefore, you should either use the form of the `SimpleDateFormat` constructor
where you pass in an explicit locale, such as `Locale.US`, or use one of the get
instance methods, or suppress this error if really know what you are doing.

> More information: 
> http://developer.android.com/reference/java/text/SimpleDateFormat.html

From the description in the JDK [documentation](https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html), it looks like the localization support in `SimpleDateFormat` is simply for providing localized letters for the pattern:

> SimpleDateFormat also supports localized date and time pattern strings. In these strings, the pattern letters described above may be replaced with other, locale dependent, pattern letters.

Note that as of Jelly Bean MR 2, there exists the [`DateFormat#getBestDateTimePattern()`](https://developer.android.com/reference/android/text/format/DateFormat.html#getBestDateTimePattern(java.util.Locale, java.lang.String)) method, which does try to provide the best localized form of a given pattern. However, we would obviously need an additional solution for lower API levels.